test/externalTests/useItemPacket.js

```diff
@@ -0,0 +1,66 @@
+const assert = require('assert')
+
+module.exports = () => async (bot) => {
+  // Test for the use_item packet format issue in 1.21+
+  // This test verifies that the correct packet format is used for different versions
+
+  const Item = require('prismarine-item')(bot.registry)
+
+  // Set up a consumable item to test with
+  await bot.test.setInventorySlot(36, new Item(bot.registry.itemsByName.bread.id, 1, 0))
+  await bot.test.becomeSurvival()
+
+  const packetsReceived = []
+
+  // Capture the use_item packet
+  const originalWrite = bot._client.write
+  bot._client.write = function (packetName, packet) {
+    if (packetName === 'use_item') {
```

Comment:
Instead of doing a trivial test checking what packets are emitted, check whether some functionality depending on handling this correctly is working